frontend/trip-planner/src/Components/Register.js

```diff
@@ -0,0 +1,85 @@
+import { useState } from "react";
+import {register} from '../Api';
+import {validateEmail} from '../utils';
+import { Link, useNavigate} from 'react-router-dom';
+import styles from './Styles/Register.css';
+
+
+function Register() {
+    const [username, setUsername] = useState('');
+    const [password, setPassword] = useState('');
+    const [nickname, setNickname] = useState('');
+    const [validUsername, setValidUsername] = useState(true);
+    const [validPassword, setValidPassword] = useState(true);
+    const [validNickname, setValidNickname] = useState(true);
+    const [registerAuth, setRegisterAuth] = useState(false);
+    const [registerSuccess, setRegisterSuccess] = useState(false);
+    const[msg, setMsg] = useState('');
+    const navigate = useNavigate();
+
+    function handleUsernameChange(e) {
+        setUsername(e.target.value);
+    };
+
+    function handlePasswordChange(e) {
+        setPassword(e.target.value);
+    };
+
+    function handleNicknameChange(e) {
+        setNickname(e.target.value);
+    };
+
+
+    async function submitForm(e) {
+        e.preventDefault();
+        setValidPassword(password.length >= 8);
+        setValidNickname(nickname.length >= 3);
+        setValidUsername(validateEmail(username));
+        
+        if ((password.length >= 8) && (nickname.length >= 3) && (validateEmail(username))) {
+            try {
+                const result = await register(username, password, nickname);
+                const jsonData = await result.json();
+                setMsg(jsonData.msg);
+                if (result.status === 201) {
+                    setRegisterSuccess(!registerSuccess);
```

Comment:
it's confusing. why not writing `setRegisterSuccess(true)`?

Reply:
also, you are not using `registerSuccess` anywhere. So you can remove this state...